contracts/SSVNetwork.sol

```diff
@@ -320,20 +321,23 @@ contract SSVNetwork is UUPSUpgradeable, Ownable2StepUpgradeable, ISSVNetwork {
             _deposit(amount);
         }
 
-        emit ValidatorAdded(msg.sender, operatorIds, publicKey, shares, cluster);
+        emit ValidatorAdded(msg.sender, operatorIds, publicKey, signatureShares, cluster);
     }
 
     function removeValidator(
         bytes calldata publicKey,
-        uint64[] memory operatorIds,
+        uint64[] calldata operatorIds,
         Cluster memory cluster
     ) external override {
-        bytes32 hashedValidator = keccak256(publicKey);
-        address validatorOwner = validatorPKs[hashedValidator].owner;
-        if (validatorOwner == address(0)) {
+        bytes32 hashedValidator = keccak256(abi.encodePacked(publicKey, msg.sender));
+        Validator memory validator = validatorPKs[hashedValidator];
+
+        if (!validator.active) {
             revert ValidatorDoesNotExist();
         }
-        if (validatorOwner != msg.sender) {
+
+        bytes32 hashedOpIds = keccak256(abi.encodePacked(operatorIds));
+        if (validatorPKs[hashedValidator].hashedOperatorIds != hashedOpIds) {
```

Comment:
This revert is more related to incorrectCluster

Reply:
@mtabasco I think Im wrong here im 99% sure its not checked in validateHashedCluster
I dont see a tie of cluster to validator im pretty sure we need exactly what you have here but with just corrected error 

if (validatorPKs[hashedValidator].hashedOperatorIds != hashedOpIds) {
revert incorrectCluster

Maybe im wrong please let me know if it is for sure checked in validateHashedCluster